client/pov-recovery/src/lib.rs

```diff
@@ -420,19 +418,17 @@ where
 
 /// Returns a stream over pending candidates for the parachain corresponding to `para_id`.
 fn pending_candidates<RC>(
-	relay_chain_client: Arc<RC>,
+	relay_chain_client: RC,
```

Comment:
```suggestion
	relay_chain_client: impl RelayChainInterface,
```

If not we can directly do this.